doc/product-mindset.md

```diff
@@ -0,0 +1,37 @@
+# Embracing the Product Mindset
+
+## Why
+
+It's important for software engineers to put on the product hat from time to time, gaining firsthand experience, insights and knowledge of our products.
+This is especially true for us, as our products - programming languages and compilers - are designed for and used by developers.
+This makes it even more important for engineers involved in day-to-day, hands-on technical work to directly participate in product roles and processes, as they are well-positioned to understand technical nuances from a user’s perspective, to study and grasp the technical details of related products, and to connect and communicate effectively with users.
+
+Benefits of doing so include:
+
+- Better decision making
+- Early detection of pain points
+- Development of better language tooling
+- Increased innovation
+- Improved empathy for users
+- Gaining broader knowledge and new perspectives
+- Personal and professional growth
+
+## How
+
+We'll introduce an additional sprint activity, alongside the two we currently have (making a release, and triaging issues and CIPs).
+
+Every sprint, an engineer is chosen, who should allocate up to 3 working days on a task in one of the following categories of their choice:
```

Comment:
We'll have to be careful not to be too ambitious with the tasks that we take on.  There are plenty of things that we might try to do that could turn out to require a lot more than three days.